src/Particular.ThroughputCollector.Persistence/IThroughputDataStore.cs

```diff
@@ -15,11 +15,14 @@ public interface IThroughputDataStore
 
     Task SaveEndpoint(Endpoint endpoint, CancellationToken cancellationToken = default);
 
-    Task RecordEndpointThroughput(EndpointIdentifier id, IEnumerable<EndpointDailyThroughput> throughput, CancellationToken cancellationToken = default);
+    Task<IDictionary<string, IEnumerable<ThroughputData>>> GetEndpointThroughputByQueueName(IEnumerable<string> queueNames, CancellationToken cancellationToken = default);
 
-    Task UpdateUserIndicatorOnEndpoints(List<Endpoint> endpointsWithUserIndicator);
+    Task RecordEndpointThroughput(string endpointName, ThroughputSource throughputSource, DateOnly date, long messageCount, CancellationToken cancellationToken = default) =>
```

Comment:
Shouldn't the method take an `EndpointIdentifier`?

Reply:
Unless I'm missing something this answer is unrelated to the question?

Reply:
EndpointIdentifier is name/throughput source.  I think throughput source is really a property of throughputdata, not the endpoint.  Removing throughputsource from the Endpoint would result in EndpointIdentifier disappearing.

Reply:
Ah ok. Well I'm not sure that it can be removed from the endpoint as the name is related to the source as well, scope could be as well, and the last collected date. Would require a fair bit of restructure.

Reply:
Maybe something we should discuss together